docs_src/microservices/application/ContextAPI.md

```diff
@@ -1,108 +1,142 @@
 
 ```go
-type Context struct {
-	// ID of the EdgeX Event (will be filled for a received JSON Event)
-	EventID string
-	
-	// Checksum of the EdgeX Event (will be filled for a received CBOR Event)
-	EventChecksum string
-	
-	// This is the ID used to track the EdgeX event through entire EdgeX framework.
-	CorrelationID string
-	
-	// OutputData is used for specifying the data that is to be outputted. Leverage the .Complete() function to set.
-	OutputData []byte
-	
-	// This holds the configuration for your service. This is the preferred way to access your custom application settings that have been set in the configuration.	
-	Configuration common.ConfigurationStruct
-	
-	// LoggingClient is exposed to allow logging following the preferred logging strategy within EdgeX.
-	LoggingClient logger.LoggingClient
-	
-	// EventClient exposes Core Data's EventClient API
-	EventClient coredata.EventClient
-	
-	// ValueDescriptorClient exposes Core Data's ValueDescriptor API
-	ValueDescriptorClient coredata.ValueDescriptorClient
-	
-	// CommandClient exposes Core Commands's Command API
-	CommandClient command.CommandClient
-	
-	// NotificationsClient exposes Support Notification's Notifications API
-	NotificationsClient notifications.NotificationsClient
-	
-	// RetryData holds the data to be stored for later retry when the pipeline function returns an error
-	RetryData []byte
-	
-	// SecretProvider exposes the support for getting and storing secrets
-	SecretProvider *security.SecretProvider
-
-	// ResponseContentType sets a custom response type
-	ResponseContentType string
+type AppFunctionContext interface {
+    CorrelationID() string
+    InputContentType() string
+    SetResponseData(data []byte)
+    ResponseData() []byte
+    SetResponseContentType(string)
+    ResponseContentType() string
+    SetRetryData(data []byte)
+    GetSecret(path string, keys ...string) (map[string]string, error)
+    SecretsLastUpdated() time.Time
+    LoggingClient() logger.LoggingClient
+    EventClient() interfaces.EventClient
+    CommandClient() interfaces.CommandClient
+    NotificationClient() interfaces.NotificationClient
+    SubscriptionClient() interfaces.SubscriptionClient
+    DeviceServiceClient() interfaces.DeviceServiceClient
+    DeviceProfileClient() interfaces.DeviceProfileClient
+    DeviceClient() interfaces.DeviceClient
+    PushToCore(event dtos.Event) (common.BaseWithIdResponse, error)
+    GetDeviceResource(profileName string, resourceName string) (dtos.DeviceResource, error)
+    AddValue(key string, value string)
+    RemoveValue(key string)
+    GetValue(key string) (string, bool)
+    GetAllValues() map[string]string
+    ApplyValues(format string) (string, error)
 }
 ```
 
+## InputContentType()
+`InputContentType()` returns the content type of the data that initiated the pipeline execution. Only useful when the TargetType for the pipeline is []byte, otherwise the data with be the type specified by TargetType.
```

Comment:
```suggestion
`InputContentType()` returns the content type of the data that initiated the pipeline execution. Only useful when the TargetType for the pipeline is []byte, otherwise the data will be the type specified by TargetType.
```